app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,21 @@
+class VotesController < ApplicationController
+  def create
+    @work = Work.find_by(id: params[:work_id])
+    @user = User.find_by(id: session[:user_id])
+
+    if !@user
```

Comment:
This logic would be great as a model method.  I suggest creating an `upvote` method in the `Work` model.